### DISCUSSION TITLE
visualizer library code source

### DISCUSSION THREAD

**Discussion by dragouf** [statelyai/xstate#1867] [Q&A]
Is the visualizer code source is available somewhere ?

I would like to display the chart in a local application.

**Comment by davidkpiano** [comment]
It is here: https://github.com/statecharts/xstate-viz

Disclaimer: The current visualizer in the above repo is _not intended for public use_, but it is open-source, so you definitely can use it wherever you want. But there are no instructions for local development; that is up to you to figure out.

**Reply by luca-vercelli** [reply]
@davidkpiano Please can you better explain this? The library xstate provides only the infrastructure for statecharts, but it does not provide any graphical representation of them?

**Reply by davidkpiano** [reply]
> @davidkpiano Please can you better explain this? The library xstate provides only the infrastructure for statecharts, but it does not provide any graphical representation of them?

Correct

**Comment by luca-vercelli** [comment]
Could you please write this, more clearly, in your project homepage?

I arrived to this project while searching for a graphic library for 
drawing diagrams. I am absolutely not interested in finite state 
machines, and I have wasted my time with this. (This message is not 
really polemic, just a suggestion to avoid other people to waste their 
time as me).



Il 14/05/2021 13:31, David Khourshid ha scritto:

**Reply by davidkpiano** [reply]
I just updated the README.md to make it clear.

Although, next time, you can look at the project name: **xstate**-viz, and maybe waste less time by understanding that it is for XState (state machines and statecharts). I take no responsibility for your time being wasted.
